### ISSUE TITLE
CMSIS-Drivers: Headers are not C++ safe

### ISSUE THREAD

**Issue by ilg-ul** [ARM-software/CMSIS#101]
The headers defining the CMSIS-Driver APIs are not C++ safe.

Suggestion:
- add `extern "C" { ... }` guards.

```
#ifdef __cplusplus
extern "C"
{
#endif

#include "Driver_Common.h"

...

#ifdef __cplusplus
}
#endif
```

**Comment by ilg-ul** [comment]
please see the patched version used in [CMSIS++](http://micro-os-plus.github.io/):

- https://github.com/xpacks/arm-cmsis/blob/xpack/CMSIS/Driver/Include/Driver_USART.h

**Comment by ReinhardKeil** [comment]
Implemented in CMSIS 5.0.0
